src/averageAges.js

```diff
@@ -1,58 +1,103 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
+ * Function returns average age of men in array.
  *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
+ * @param {object[]} people array of people.
+ * @param {number=} [century] if this parameter is speficied then the
+ * function calculates average age only for men who died in this century.
  *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
+ * @return {number} the average age of men.
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century === undefined
+    ? people.filter(person => person.sex === 'm')
+    : people
+      .filter(person => person.sex === 'm')
+      .filter(person => Math.ceil(person.died / 100) === century);
+
+  const result = men
+    .map(person => person.died - person.born)
+    .reduce((accumulator, current) => accumulator + current) / men.length;
+
+  return Math.round(result * 100) / 100;
```

Comment:
```suggestion
  return result;
```
Have no idea why would you multiplying your result by 100, then dividing it by 100 back, and then rounding it to return...
Your variable 'result' - is actually what we need